ThirdHomework21.09.21/BinSearch/BinSearch.c

```diff
@@ -0,0 +1,253 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <time.h>
+
+void swap(int* firstVariable, int* secondVariable)
+{
+    *firstVariable ^= *secondVariable;
+    *secondVariable ^= *firstVariable;
+    *firstVariable ^= *secondVariable;
+}
+
+void fillArrayWithRandomElements(int theArray[], int lengthOfTheArray, int mod)
+{
+    srand((unsigned)time(NULL));
+    for (int i = 0; i < lengthOfTheArray; ++i)
+    {
+        theArray[i] = rand() % mod;
+    }
+}
+
+void printArray(const int theArray[], const int lengthOfTheArray)
+{
+    for (int i = 0; i < lengthOfTheArray; ++i)
+    {
+        printf("%d ", theArray[i]);
+    }
+    printf("\n\n");
+}
+
+void insertionSort(int theArray[], int lengthOfTheArray)
```

Comment:
```suggestion
void insertionSort(int theArray[], const int lengthOfTheArray)
```
Раз уж Вы везде const-ы ставите, то и тут можно